File: Cliente RestAPI/Cliente-frontend/src/service/ProductoService.js
import { axiosFonda } from "./AuthService";
const PRODUCTOS_API = 'http://localhost:7071/api/producto';

export const listProductos = () => axiosFonda.get(PRODUCTOS_API);

export const listProductosActivos = () => axiosFonda.get(`${PRODUCTOS_API}/activos`);


export const crearProductos = (producto) => {
  return axiosFonda.post(PRODUCTOS_API, producto, {
    headers: { "Content-Type": "application/json" },
  });
}
export const getProducto = (idProducto) => axiosFonda.get(`${PRODUCTOS_API}/${idProducto}`)

export const updateProducto = (idProducto, producto) => {
  return axiosFonda.put(`${PRODUCTOS_API}/${idProducto}`, producto, {
    headers: { "Content-Type": "application/json" },
  });
}

export const deleteProducto = (idProducto) => axiosFonda.delete(`${PRODUCTOS_API}/${idProducto}`)

export const softDeleteProducto = (idProducto) => axiosFonda.put(`${PRODUCTOS_API}/ocultar/${idProducto}`)

export const activarProducto = (idProducto) => axiosFonda.put(`${PRODUCTOS_API}/activar/${idProducto}`);

export const getProductosInactivos = () => axiosFonda.get(`${PRODUCTOS_API}/inactivos`);

export const subirImagen = (idProducto, formData) => {
  return axiosFonda.post(`${PRODUCTOS_API}/uploadImage/${idProducto}`, formData);
};

export const buscarPorTipo = (nombreTipo) => axiosFonda.get(`${PRODUCTOS_API}/buscarPorTipo?nombreTipo=${nombreTipo}`);


export const buscarEntrePrecios = (min, max) => axiosFonda.get(`${PRODUCTOS_API}/buscarEntrePrecios?min=${min}&max=${max}`);

export const BuscarProducto = (nombreProducto) => axiosFonda.get(`${PRODUCTOS_API}/buscar?nombreProducto=${nombreProducto}`);

File: Fonda/src/main/java/itch/fonda/service/ProductoService.java
package itch.fonda.service;
import java.util.List;

import itch.fonda.dto.ProductoDto;

public interface ProductoService {
	ProductoDto createProducto (ProductoDto productoDto);
		
	//OCULTOS
    void softDeleteProducto(Integer idProducto);
    //Ver de Nuevo el producto
    void restoreProducto(Integer idProducto); 
    
    //LISTA DE PRODUCTOS INACTIVOS
    List<ProductoDto> getProductosInactivos();
	
	//BUSCAR Producto POR ID
	ProductoDto getProductoById(Integer Id);
	
	//OBTENER TODOS LOS Producto
	List<ProductoDto> getAllProducto();
	
	//OBTENER TODOS LOS Producto Activos
	List<ProductoDto> getAllProductosActivos();
	
	//CONTRUIR REST API UPDATE Producto
	ProductoDto updateProducto(Integer Id, ProductoDto updateProducto);
	
	//CONSTRUIR DELETE REST API Producto
	void deleteProducto(Integer Id); 
	
	//Guardar Imagen
	void guardarNombreFoto(Integer id, String nombreFoto);
	
	//Actualizar foto
	void actualizarNombreImagen(Integer IdProducto, String nombreFoto);
	
	//BUSQUEDAS
	List<ProductoDto> buscarPrecioEntre(Double min, Double max);
	List<ProductoDto> buscarPorTipo(String nombreTipo);
	List<ProductoDto> buscarNombreContiene(String nombreProducto);
}
